### DISCUSSION TITLE
[Feature Request] Wear OS Support

### DISCUSSION THREAD

**Discussion by hpnightowl** [ente-io/ente#485] [Enhancements]
- Wear OS Instant Preview of Codes from Wrist only

**Comment by errorgurumeditation** [comment]
would also be good to add it for watchos (apple watch)

**Reply by muesli79** [reply]
Wear OS would be great!

**Comment by AxelTerizaki** [comment]
I was interested in switching to Ente, but alas, having an Apple Watch I'm way too used to have my 2FA codes at my wrist, easily reachable by pressing a few buttons.

So yes, a wearable version would be really nice.

**Comment by TheMikeyR** [comment]
I've moved from Authy to Ente Auth do to all the recent Authy leaks.. 
I'm really missing having an apple watch app to find my codes.. Hope it will be added to the roadmap.

**Comment by notDavid** [comment]
Hi there, please consider moving this discussion to an "issue", so people can then follow it and get notified when the status changes to closed, etc.

Thank you!

**Reply by AxelTerizaki** [reply]
It was an issue before it got moved here, actually : https://github.com/ente-io/ente/issues/143

**Reply by mnvr** [reply]
@notDavid You can watch discussions too. 

<img width="299" alt="Screenshot 2024-07-08 at 09 33 44" src="https://github.com/ente-io/ente/assets/24503581/2e02cbd0-eab1-40fb-9e01-1a0dd9ea7f7f">


The intent was that issues can be dedicated to bug fixes, whilst feature request and other improvements can be tracked in discussions.

**Reply by notDavid** [reply]
> @notDavid You can watch discussions too.

I'm aware of that, but that will give you a notification (e-mail) for every single comment.
Only with issues, you can set `Notifications > Customize > Status` to only get a notification when the issue is actually released (closed).

> The intent was that issues can be dedicated to bug fixes, whilst feature request and other improvements can be tracked in discussions.

I see, thank you. I just wanted to mention it for your consideration. Not sure if it's recommended/customary [to use discussions, instead of issues, for feature requests](https://docs.github.com/en/discussions/guides/best-practices-for-community-conversations-on-github#learning-about-conversations-on-github) 🤷🏼‍♂️ 🤔

**Reply by vishnukvmd** [reply]
> Not sure if it's recommended/customary [to use discussions, instead of issues, for feature requests](https://docs.github.com/en/discussions/guides/best-practices-for-community-conversations-on-github#learning-about-conversations-on-github) 🤷🏼‍♂️ 🤔

Again, this is intentional. Also, the link you shared has this to say:
```
GitHub Discussions is useful for teams in a number of scenarios.
- You want to collect feedback from a wider community about a project
- You want to keep bug fixes, feature requests, and general conversations separate
- You want to measure interest in new ideas and gauge community opinion using polls
```

All of these apply.

**Comment by SimStm** [comment]
I started using the Ente Auth app recently and love the fact that I could access anywhere, from my Android phone to my Windows and Mac devices in a safe way. It would be great to have the app also in `Wear OS` and `watchOS` devices for fast access. :)

**Comment by bruceoberg** [comment]
i would also like an ente auth apple watch app. with no issues for this, it's hard to tell if the dev team is working on it or not. perhaps there's a roadmap somewhere? i coudn't find one.

i'm a developer with spare time and was considering taking a swing at adding an apple watch app to the auth codebase. i've found two guides that show how it could be done: [one from scott hatfield](https://medium.com/@Toglefritz/building-a-watchos-companion-app-for-flutter-08f80ea8cb1b) and [another from cheesecake labs](https://cheesecakelabs.com/blog/flutter-apps-apple-watch-integration/).

if i work on this, what are the chances that a PR would be accepted? the docs (understandably) don't encourage PRs from outsiders. but i want this feature and would like to help make it happen.

**Reply by vishnukvmd** [reply]
Hey, if the feature is free from errors we'll do our best to get the PR merged.

**Reply by bruceoberg** [reply]
ok thanks. i’m currently blocked by flutter. ente’s version breaks watch integration for ios apps. i believe it’s fixed in flutter 3.32, but i’m told that ente can’t upgrade to 3.32 because of other bugs.

**Reply by vishnukvmd** [reply]
Yes, 3.32 broke a bunch of other things so we ended up having to downgrade. Hopefully there'll be a more stable (less unstable?) version we can switch to soon.

**Comment by VslVictor7** [comment]
Hi, is implementing ente auth in the apple watch still in the roadmap? Is there an update if its ever going to be a thing?

**Reply by bruceoberg** [reply]
the last time i checked (november 2025), flutter was still blocking watch integration.

flutter is the google toolset that ente uses to build a cross platform app. the xcode project files generated by flutter appear to prevent a watch companion app from being created by xcode itself. and flutter doesn't appear to be able to add apple watch support on its own.

i'll check again this month, but it's hard to say if/when apple watch support will be "easy" to do. i dearly want the feature (because authy's apple watch app is crummy), but making it happen is relatively difficult for a volunteer like me to do.

furthermore, i'm guessing flutter has some sort of wear os support, and it's not clear that wear-os and apple-watch code could share any source at all (like they do for android/ios).

**Reply by bruceoberg** [reply]
with [the latest ente commit](https://github.com/ente-io/ente/commit/c3be0eb2132350eca2a5f845e752b8827664c875) and xcode version 26.2 (17C52), i've verified that adding an apple watch companion app to the ente auth app is still broken. in ente auth's xcode project, you can create a new Watch companion app target. but within that target, xcode does not allow you to create a watchOS destination. and without a watchOS destination, you can't build anything.

it appears that there is still something in the (flutter generated) xcode project file that convinces xcode that it can't add a watchOS destination to a watchOS target. i have not searched the flutter forums for anything related.

**Reply by bruceoberg** [reply]
for the record, here is [a tutorial from amorn apichattanakul](https://medium.com/kbtg-life/adding-apple-watch-to-flutter-app-via-flutter-method-channel-f1443532d94e) ([with source](https://github.com/theamorn/flutter-apple-watch)) about connecting flutter iOS apps with their watchOS counterparts.

**Reply by VslVictor7** [reply]
I see, tysm for clarifying the situation. Of course i didn't mrsn to be annoying hahah but thank you nonetheless.

Quite sad it's something outside of the project's control.

**Reply by bruceoberg** [reply]
well well well. i'm old school, not really an AI/LLM guy (yet), but i took a chance and had claude look at the project/target problem. to my great surprise, it (he?) sussed out what was going on and i now have a project with an apple watch target.

there's still a bunch of work to do, but i am no longer blocked. i should be able to make progress over the next few months.

**Reply by VslVictor7** [reply]
That's awesome! I wanted to help as well but my knowledge in flutter is really bad hahah

**Reply by AShuayto** [reply]
@bruceoberg be the hero we all need! any progress on this? I'm no swift pro but if you need help let me know!

**Reply by bruceoberg** [reply]
i am still trying to figure out how to build the ios auth app from source (#8678). there's magic involved that is not written down anywhere. it's funny that all the roadblocks are build related. i believe the app itself will be simple - it will map almost 1-1 with how the apple watch sample app works.

**Reply by AShuayto** [reply]
Where's our friend Claude?? Haha

**Comment by sepsol** [comment]
3 years since the issue was created, I still am looking for a solution that supports WearOS.
